src/SprykerEco/Zed/Unzer/Business/Checkout/UnzerPostSaveCheckoutHook.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Checkout;
+
+use Generated\Shared\Transfer\CheckoutResponseTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Shared\Unzer\UnzerConfig as SharedUnzerConfig;
+use SprykerEco\Zed\Unzer\Business\Checkout\Mapper\UnzerCheckoutMapperInterface;
+use SprykerEco\Zed\Unzer\Business\Payment\ProcessorResolver\UnzerPaymentProcessorStrategyResolverInterface;
+use SprykerEco\Zed\Unzer\Business\Payment\Saver\UnzerPaymentSaverInterface;
+use SprykerEco\Zed\Unzer\UnzerConfig;
+use SprykerEco\Zed\Unzer\UnzerConstants;
+
+class UnzerPostSaveCheckoutHook implements UnzerCheckoutHookInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Checkout\Mapper\UnzerCheckoutMapperInterface
+     */
+    protected $unzerCheckoutHookMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\UnzerConfig
+     */
+    protected $unzerConfig;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Payment\Saver\UnzerPaymentSaverInterface
+     */
+    protected $unzerPaymentSaver;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Payment\ProcessorResolver\UnzerPaymentProcessorStrategyResolverInterface
+     */
+    protected $paymentProcessorStrategyResolver;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\Checkout\Mapper\UnzerCheckoutMapperInterface $unzerCheckoutHookMapper
+     * @param \SprykerEco\Zed\Unzer\UnzerConfig $unzerConfig
+     * @param \SprykerEco\Zed\Unzer\Business\Payment\Saver\UnzerPaymentSaverInterface $unzerPaymentSaver
+     * @param \SprykerEco\Zed\Unzer\Business\Payment\ProcessorResolver\UnzerPaymentProcessorStrategyResolverInterface $paymentProcessorStrategyResolver
+     */
+    public function __construct(
+        UnzerCheckoutMapperInterface $unzerCheckoutHookMapper,
+        UnzerConfig $unzerConfig,
+        UnzerPaymentSaverInterface $unzerPaymentSaver,
+        UnzerPaymentProcessorStrategyResolverInterface $paymentProcessorStrategyResolver
+    ) {
+        $this->unzerCheckoutHookMapper = $unzerCheckoutHookMapper;
+        $this->unzerConfig = $unzerConfig;
+        $this->unzerPaymentSaver = $unzerPaymentSaver;
+        $this->paymentProcessorStrategyResolver = $paymentProcessorStrategyResolver;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponse
+     *
+     * @return void
+     */
+    public function execute(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse): void
+    {
+        if ($quoteTransfer->getPayment() === null) {
+            return;
+        }
+
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== SharedUnzerConfig::PROVIDER_NAME) {
```

Comment:
```suggestion
        if ($quoteTransfer->getPaymentOrFail()->getPaymentProvider() !== SharedUnzerConfig::PROVIDER_NAME) {
```
same for other cases